util/pkg/vfs/s3fs.go

```diff
@@ -155,6 +163,50 @@ func (p *S3Path) WriteFile(data io.ReadSeeker, aclObj ACL) error {
 	return nil
 }
 
+func (p *S3Path) GetVersioning() (bool, error) {
+	client, err := p.client()
+	if err != nil {
+		return false, err
+	}
+
+	request := &s3.GetBucketVersioningInput{}
+	request.SetBucket(p.bucket)
+
+	response, err := client.GetBucketVersioning(request)
+	if err != nil {
+		return false, fmt.Errorf("error getting s3 bucket versioning: %v", p, err)
+	}
+
+	if strings.Contains(response.String(),"Status: \"Enabled\"") {
+		return true, nil
+	} else {
+		return false, nil
+	}
+}
+
+func (p *S3Path) SetVersioning(status string) ( error) {
```

Comment:
Please add godocs here. We will need to expose this through the Interface, but we need to figure out how to do it first.  Let's just log first.

Reply:
Adding to interface means to implement it for other storage types too. This code is only for S3. Do you want me to add empty functions in other types too?